regparser/layer/def_finders.py

```diff
@@ -0,0 +1,125 @@
+# vim: set fileencoding=utf-8
+"""Parsers for finding a term that's being defined within a node"""
+from itertools import chain
+import re
+
+from regparser.citations import Label
+from regparser.grammar import terms as grammar
+import settings
+
+
+class Ref(object):
+    def __init__(self, term, label, start):
+        self.term = unicode(term).lower()
+        self.label = label
+        self.start = start
+        self.end = self.start + len(term)
+        self.position = (self.start, self.end)
+
+    def __eq__(self, other):
+        """Equality depends on equality of the fields"""
+        return isinstance(other, Ref) and repr(self) == repr(other)
+
+    def __repr__(self):
+        return 'Ref( term=%s, label=%s, start=%s )' % (
+            repr(self.term), repr(self.label), repr(self.start))
+
+
+class ExplicitIncludes(object):
+    """Definitions can be explicitly included in the settings"""
+    def find(self, node):
+        refs = []
+        cfr_part = node.label[0] if node.label else None
+        for included_term, context in settings.INCLUDE_DEFINITIONS_IN.get(
+                cfr_part, []):
+            if context in node.text and included_term in node.text:
+                pos_start = node.text.index(included_term)
+                refs.append(Ref(included_term, node.label_id(), pos_start))
+        return refs
+
+
+class SmartQuotes(object):
```

Comment:
@khandelwal pointed me to Python's Abstract Base Classes. Very interesting; I'll implement